### PR TITLE
Update to symfony/console 6.0 and newer version of phpunit

### PR DESCRIPTION
**Pull request by cjobeili**
This PR aims to update symfony/console version in order to make it work for brefphp/bref in ref of #60 

I have updated to a newer version of phpunit in a separated commit too

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -15,13 +15,13 @@
...
-        "php": ">=7.0",
-        "symfony/console": "~3.0|~4.0|~5.0",
+        "php": ">=7.4",
+        "symfony/console": "~3.0|~4.0|~5.0|~6.0",
         "php-di/invoker": "~2.0",
         "psr/container": "^1.0|^2.0"
     },
     "require-dev": {
-        "phpunit/phpunit": "~6.4",
+        "phpunit/phpunit": ">=6.4",
```
Comment (mnapoli) [inline composer.json]: This runs the risk of breaking with new PHPUnit versions. Maybe try something like `^6.4 || 7 || 8`?

**Comment by mnapoli** [comment]
🤦  thank you Travis-CI… we have no CI anymore.

**Comment by osteel** [comment]
Hi!

What's preventing this from moving forward? Since Valet uses Silly as a dependency, it seems that every local environment featuring Valet is currently prevented from upgrading to `symfony/console` 6.

Cheers

**Comment by mnapoli** [comment]
Honestly, it depends on our degree of confidence here since we have no CI.

The ideal scenario is someone sends a PR to add GitHub Actions CI, but I know that this takes time (as I won't have much time in the near future to tackle that). If this PR is really blocking, and you both are confident this PR is good, I'm fine with merging. I don't want to be the SPOF here.

Also there's this minor comment that needs to be adjusted before merging: https://github.com/mnapoli/silly/pull/61#discussion_r773110496

Any help is welcome!

**Comment by osteel** [comment]
Ok cool. Can't promise anything but I may try and look into adding GitHub CI instead (in a separate PR)

**Comment by cjobeili** [comment]
> Also there's this minor comment that needs to be adjusted before merging: [#61 (comment)](https://github.com/mnapoli/silly/pull/61#discussion_r773110496)

I've amend the last commit with the changes suggested ! Thanks and sorry for the delay !

**Comment by osteel** [comment]
Sweet, thanks @cjobeili. I think a good approach now would be to fix CI in a different PR, get that other PR merged in, then merge `master` back into this PR and make sure CI passes. Does that sound sensible?

**Comment by cjobeili** [comment]
> Sweet, thanks @cjobeili. I think a good approach now would be to fix CI in a different PR, get that other PR merged in, then merge `master` back into this PR and make sure CI passes. Does that sound sensible?

Not at all, i've just worked on github actions so https://github.com/cjobeili/silly/actions/runs/1813643403 .
Here you can see tests for 7.0 to 8.1 in a matrix (with 8+ failing due to composer file). I will make a PR for this.

**Comment by mnapoli** [comment]
I've merged #64

**Comment by cjobeili** [comment]
Updated with master and kaboom ! I'm looking at it

**Comment by cjobeili** [comment]
So here my "investigations":

7.0, 7.1 and 7.2 fails because of the new composer condition added >=7.4
8.1 and 8.0 fails because of the version of phpunit , i need to add `|| 9` i think
and i've missed 7.4 on the github workflow.

Are we ok to drop all version before 7.4 ?

Thanks!

**Comment by mnapoli** [comment]
Yes!

**Comment by cjobeili** [comment]
The last fail is due to `--prefer-lowest` composer arg.
For 8.1 it tried to install phpunit in version 8.5.12 and fails because of : 

`Run vendor/bin/phpunit
PHP Fatal error:  Cannot acquire reference to $GLOBALS in /home/runner/work/silly/silly/vendor/phpunit/phpunit/src/Util/Configuration.php on line 40[7](https://github.com/cjobeili/silly/runs/5122961894?check_suite_focus=true#step:7:7)
Error: Process completed with exit code 255.`

this bug has been fixed since [8.5.14](https://github.com/sebastianbergmann/phpunit/blob/master/ChangeLog-8.5.md#8514---2021-01-17) [commit](https://github.com/sebastianbergmann/phpunit/issues/4575) but composer seems to like this version.

Any idea how to handle this issue ?

Thanks!

**Comment by osteel** [comment]
@cjobeili this may be a naive answer but couldn't you just change the dependency requirement for this?

```
"phpunit/phpunit": "^6.4|^7|^8.5.14|^9",
```

Since PHPUnit 8 is introduced for the first time anyway.

Thanks for doing this by the way!

**Comment by PieterCappelle** [comment]
Please update & merge this.

**Comment by mnapoli** [comment]
@PieterCappelle feel free to send a PR.

**Comment by mnapoli** [comment]
@cjobeili I'm fine to merge this, is this ready to merge for you?

**Comment by cjobeili** [comment]
@mnapoli I'm fine with this, i've drop the low-deps matrix because it was a pain to make it work "quickly". If it's ok for you it's ok for me ! All tests passes with standard deps.

**Comment by mnapoli** [comment]
Thank you!

**Pull request merged**